Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -32,41 +38,55 @@ public Block(CustomUIPane pane) {
         try {
             output = new OutputAnchor(this, pane);
         } catch (IOException e) {
-            // TODO Find a good way to handle this
             e.printStackTrace();
         }
 
-        parentPane.selectedBlockProperty().addListener(event -> {
-            if (parentPane.getSelectedBlock().isPresent() && this.equals(parentPane.getSelectedBlock().get())) {
-                this.getStyleClass().add("selected");
-            } else {
-                this.getStyleClass().removeAll("selected");
-            }
-        });
+        parentPane.selectedBlockProperty()
+                .addListener(
+                        event -> {
+                            if (parentPane.getSelectedBlock().isPresent()
+                                    && this.equals(parentPane
+                                            .getSelectedBlock().get())) {
+                                this.getStyleClass().add("selected");
+                            } else {
+                                this.getStyleClass().removeAll("selected");
+                            }
+                        });
+
+        this.addEventHandler(MouseEvent.MOUSE_CLICKED, this::handleMouseEvent);
 
-        this.addEventHandler(MouseEvent.MOUSE_CLICKED, this::select);
+        this.createContextMenu();
     }
 
-    /** Sets this block as the selected block. */
-    private void select(MouseEvent mouseEvent) {
-        parentPane.setSelectedBlock(this);
+    private void createContextMenu() {
+        contextMenu = new CircleMenu(this);
     }
 
-    /**
-     * @return the output Anchor for this Block
-     */
+    /** Returns the output Anchor for this Block. */
     public final OutputAnchor getOutputAnchor() {
         return output;
     }
 
-    /** @return the parent pane of this Component. */
+    /**
+     * Sets this block as the selected block. When this block has already been
+     * selected it spawns a contextMenu instead.
+     */
+    private void handleMouseEvent(MouseEvent t) {
+        if (parentPane.getSelectedBlock().isPresent()
+                && parentPane.getSelectedBlock().get().equals(this)
+                && t.getButton().equals(MouseButton.PRIMARY)) {
```

Comment:
I have no reason to believe it will not since mouse events are cased into touch events. I will however need to test this later this week.